### PR TITLE
SCRUM-4538 upgrade to independent genetic modifier model

### PR DESCRIPTION
**Pull request by cmpich**

### CODE REVIEW

File: agr_indexer/src/main/java/org/alliancegenome/indexer/indexers/curation/DiseaseAnnotationCurationIndexer.java [hunk 1]
```diff
@@ -399,10 +376,23 @@ private void populateBaseDiseaseAnnotationDocument(BiologicalEntity biologicalEn
 	}
 
 	private static String getGeneticModifierConsolidatedKey(DiseaseAnnotation da) {
-		if (da.getDiseaseGeneticModifierRelation() == null && CollectionUtils.isEmpty(da.getDiseaseGeneticModifiers())) {
+		if (da.getDiseaseGeneticModifierRelation() == null &&
```
Comment (devopsagr) [inline agr_indexer/src/main/java/org/alliancegenome/indexer/indexers/curation/DiseaseAnnotationCurationIndexer.java:379]: 🚫 **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.whitespace.OperatorWrapCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>'&&' should be on a new line.

File: agr_indexer/src/main/java/org/alliancegenome/indexer/indexers/curation/DiseaseAnnotationCurationIndexer.java [hunk 2]
```diff
@@ -399,10 +376,23 @@ private void populateBaseDiseaseAnnotationDocument(BiologicalEntity biologicalEn
...
+			CollectionUtils.isEmpty(da.getDiseaseGeneticModifierAlleles()) &&
```
Comment (devopsagr) [inline agr_indexer/src/main/java/org/alliancegenome/indexer/indexers/curation/DiseaseAnnotationCurationIndexer.java:380]: 🚫 **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.whitespace.OperatorWrapCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>'&&' should be on a new line.

File: agr_indexer/src/main/java/org/alliancegenome/indexer/indexers/curation/DiseaseAnnotationCurationIndexer.java [hunk 3]
```diff
@@ -399,10 +376,23 @@ private void populateBaseDiseaseAnnotationDocument(BiologicalEntity biologicalEn
...
+			CollectionUtils.isEmpty(da.getDiseaseGeneticModifierGenes()) &&
```
Comment (devopsagr) [inline agr_indexer/src/main/java/org/alliancegenome/indexer/indexers/curation/DiseaseAnnotationCurationIndexer.java:381]: 🚫 **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.whitespace.OperatorWrapCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>'&&' should be on a new line.

File: agr_indexer/src/main/java/org/alliancegenome/indexer/indexers/curation/DiseaseAnnotationCurationIndexer.java [hunk 4]
```diff
@@ -441,8 +431,19 @@ private static String getConsolidationKey(DiseaseAnnotation da, String relation)
 	}
 
 	private static void populateGeneticModifier(DiseaseAnnotation da, DiseaseAnnotationDocument adad) {
-		if (CollectionUtils.isNotEmpty(da.getDiseaseGeneticModifiers())) {
-			List<BiologicalEntity> geneticModifiers = da.getDiseaseGeneticModifiers().stream().filter(Objects::nonNull).toList();
+		if (CollectionUtils.isNotEmpty(da.getDiseaseGeneticModifierAlleles()) ||
```
Comment (devopsagr) [inline agr_indexer/src/main/java/org/alliancegenome/indexer/indexers/curation/DiseaseAnnotationCurationIndexer.java:434]: 🚫 **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.whitespace.OperatorWrapCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>'||' should be on a new line.

File: agr_indexer/src/main/java/org/alliancegenome/indexer/indexers/curation/DiseaseAnnotationCurationIndexer.java [hunk 5]
```diff
@@ -441,8 +431,19 @@ private static String getConsolidationKey(DiseaseAnnotation da, String relation)
...
+			CollectionUtils.isNotEmpty(da.getDiseaseGeneticModifierGenes()) ||
```
Comment (devopsagr) [inline agr_indexer/src/main/java/org/alliancegenome/indexer/indexers/curation/DiseaseAnnotationCurationIndexer.java:435]: 🚫 **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.whitespace.OperatorWrapCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>'||' should be on a new line.

File: pom.xml [hunk 6]
```diff
@@ -32,7 +32,7 @@
 		<quarkus.platform.artifact-id>quarkus-bom</quarkus.platform.artifact-id>
 		<quarkus.platform.group-id>io.quarkus.platform</quarkus.platform.group-id>
 		<quarkus.platform.version>${quarkus.version}</quarkus.platform.version>
-		<curation.version>v0.34.2</curation.version>
+		<curation.version>v0.35.0-rc2</curation.version>
```
Comment (oblodgett) [inline pom.xml]: This is not production code... it needs to be `v0.35.0`

Comment (cmpich) [reply pom.xml]: Version v0.35.0 does not exist in maven central yet.

**Review by oblodgett — APPROVED** [review]

**Pull request merged**